core/src/replay_stage.rs

```diff
@@ -1425,44 +1583,37 @@ impl ReplayStage {
             }
             assert_eq!(*bank_slot, bank.slot());
             if bank.is_complete() {
-                if !blockstore.has_duplicate_shreds_in_slot(bank.slot()) {
-                    bank_progress.replay_stats.report_stats(
-                        bank.slot(),
-                        bank_progress.replay_progress.num_entries,
-                        bank_progress.replay_progress.num_shreds,
-                    );
-                    did_complete_bank = true;
-                    info!("bank frozen: {}", bank.slot());
-                    if let Some(transaction_status_sender) = transaction_status_sender.clone() {
-                        transaction_status_sender.send_transaction_status_freeze_message(&bank);
-                    }
```

Comment:
This if-statement wasn't in the original delete block, so I added it to the addition block below